src/effects/effectslot.cpp

```diff
@@ -386,6 +365,15 @@ void EffectSlot::clear() {
     for (const auto& pParameter : m_buttonParameters) {
         pParameter->clear();
     }
+
+    m_parametersById.clear();
+    for (int i = 0; i < m_parameters.size(); ++i) {
+        EffectParameter* pParameter = m_parameters.at(i);
+        m_parameters[i] = NULL;
+        delete pParameter;
+    }
+
+    removeFromEngine();
```

Comment:
Should EffectSlot::removeFromEngine ever be called from anywhere else? If not, maybe that function should be removed and its code moved into this function.

Reply:
No, It is not being called from anywhere else. But I think we should keep it as it is to maintain the uniformity of having a add/remove to/from engine function in each effect layer.